components/NavButton/NavButton.js

```diff
@@ -1,15 +1,21 @@
 import * as Styled from "./NavButton.styled";
+import { useTheme } from "styled-components";
 
 export default function NavButton({
   handleClick,
   disabled,
   children,
   linkToPage,
 }) {
+  const theme = useTheme();
   return (
-    <Styled.Button onClick={handleClick} $disabled={disabled}>
+    <Styled.Button onClick={handleClick} $disabled={disabled} theme={theme}>
       {linkToPage ? (
-        <Styled.ButtonTextLink $disabled={disabled} href={linkToPage}>
+        <Styled.ButtonTextLink
+          $disabled={disabled}
```

Comment:
if disable is a boolean i am pretty sure you can just writ $disabled as prop and don't need the ={disabled}